src/Sylius/Bundle/ResourceBundle/DependencyInjection/Compiler/ObjectToIdentifierServicePass.php

```diff
@@ -46,6 +49,11 @@ public function process(ContainerBuilder $container)
             if (!$container->hasDefinition('sylius_entity_to_identifier')) {
                 $container->setAlias('sylius_entity_to_identifier', 'sylius_document_to_identifier');
             }
+
+            if (!$container->hasDefinition('doctrine')) {
+                $definition = $container->findDefinition('sylius.form.type.entity_hidden');
+                $definition->replaceArgument(0, new Reference('doctrine_mongodb'));
```

Comment:
Personally I start to prefer smaller PRs as @pjedrzejewski is not having time to review & we wait way to long for merges, so this can be changed in another PR IMO.